File: Leetcode_Python/implement_strStr.py
class Solution:
    # @param haystack, a string
    # @param needle, a string
    # @return an integer
    def strStr(self, haystack, needle):
        hl = len(haystack)
        nl = len(needle)
        if hl < nl:
            return -1
        if nl == 0:
            return 0
        i = 0
        while i < (hl - nl + 1):
            j = 0
            while j < nl:
                if needle[j] != haystack[i+j]:
                    i += 1
                    break
                else:
                    j += 1
                    if j == nl:
                        return i
        return -1

File: cleanCode50/ImplementStrStr.java
public class ImplementStrStr {
    public int strStr(String haystack, String needle) {
        if (haystack == null || needle == null || needle.length() > haystack.length()) {
            return -1;
        }
        int lengthHaystack = haystack.length();
        int lengthNeedle = needle.length();
        if (lengthNeedle == 0) {
            return 0;
        }
        for (int i = 0; i <= lengthHaystack - lengthNeedle; i++) {
            int j = 0;
            while (j < lengthNeedle) {
                if (haystack.charAt(i + j) != needle.charAt(j)) {
                    break;
                } else {
                    j++;
                }
                if (j == lengthNeedle) {
                    return i;
                }
            }
        }
        return -1;
    }
}